### PR TITLE
UI: Sticky Top

### LINKED ISSUE #5035: Patch Page: "Freeze" top of page
I'm not too big on HTML, so I'm not sure about the proper terminology.

It would be cool if on the patch page (could also work well on other pages ) the top area could be "frozen". By this I mean that the area remains visible even if you scroll down. 
This way you don't have to scroll to the top to stage or refresh.

image

Thoughts?

### PR DESCRIPTION
**Pull request by mbtools**
With this change, the top bar will become sticky. Meaning, it remains on the top of the page when scrolling down.

Since IE does not support the "sticky" CSS feature, it's based on CSS + JS. It should work on SAPGUI win and web.
Issues:

- There's a bit of a wiggle when scrolling begins 
- The alignment is not the best on the diff page (which has full width). 
- Jumping to files on the diff page puts the filename to the top which is under the sticky bar

Maybe some of you CSS wizards can figure out how to fix these.

Closes #5035

### CODE REVIEW

**Review by larshp — DISMISSED** [review]

File: src/ui/zabapgit_js_common.w3mi.data.js [hunk 1]
```diff
@@ -1,2327 +1,2355 @@
...
-  window.sessionStorage.setItem("scrollTop", 0);
-}
-
-function memoizeScrollPosition(fn){
-  return function(){
-    saveScrollPosition();
-    return fn.call(this, fn.args);
-  }.bind(this);
-}
+/**********************************************************
```
[the remark this answers is not in the document]
Comment (larshp) [reply src/ui/zabapgit_js_common.w3mi.data.js:1]: @sbcgua any thoughts regarding the line endings?

[the remark this answers is not in the document]
Comment (sbcgua) [reply src/ui/zabapgit_js_common.w3mi.data.js:1]: I think the problem is that the repo does not have `.gitattributes`. I had some issues like that before with other repos.

Just submitted #5081. But this is bit dark magic. What I would try is:
1) merge #5081
2) rebase this PR onto master
3) And after this (supposing `.gitattributes` appered in the root folder) follow https://dev.to/deadlybyte/please-add-gitattributes-to-your-git-repository-1jld in the command line to be sure everything was reset.
4) force push

Make sense to review the diff in the command line or with some tool. But this is something I would try. Without 100% confidence though.

[the remark this answers is not in the document]
Comment (sbcgua) [reply src/ui/zabapgit_js_common.w3mi.data.js:1]: And also this: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings
`git config --global core.autocrlf true`
It should help keeping LFs in the working copy in windows

**Comment by larshp** [comment]
@mbtools check conflicts

**Comment by larshp** [comment]
image

**Pull request merged**